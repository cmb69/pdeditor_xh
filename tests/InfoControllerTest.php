<?php

namespace Pdeditor;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    /** @var SystemChecker */
    private $systemChecker;

    /** @var View */
    private $view;

    public function setUp(): void
    {
        $this->systemChecker = new FakeSystemChecker();
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["pdeditor"]);
    }

    private function sut(): InfoController
    {
        return new InfoController("./plugins/pdeditor/", $this->systemChecker, $this->view);
    }

    public function testRendersPluginInfo(): void
    {
        $response = $this->sut()();
        $this->assertSame("Pdeditor 2.1-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
