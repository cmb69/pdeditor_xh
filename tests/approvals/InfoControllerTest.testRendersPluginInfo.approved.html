<h1>Pdeditor 1.1-dev</h1>
<h2>System check</h2>
<p class="xh_fail">PHP version ≥ 7.1.0: not okay</p>
<p class="xh_fail">Extension &quot;pcre&quot; loaded: not okay</p>
<p class="xh_fail">Extension &quot;spl&quot; loaded: not okay</p>
<p class="xh_fail">CMSimple_XH version ≥ 1.7.0: not okay</p>
<p class="xh_warning">Folder &quot;./plugins/pdeditor/css/&quot; writable: not okay</p>
<p class="xh_warning">Folder &quot;./plugins/pdeditor/languages/&quot; writable: not okay</p>
